solver/src/driver.rs

```diff
@@ -285,23 +286,45 @@ impl Driver {
         }
     }
 
-    // Record metric with the amount of orders that were matched but not settled in this runloop (effectively queued for the next one)
-    // Should help us to identify how much we can save by parallelizing execution.
-    fn report_matched_but_unsettled_orders(
+    /// Record metrics on the matched orders from a single batch. Specifically we report on
+    /// the number of orders that were;
+    ///  - matched but not settled in this runloop (effectively queued for the next one)
+    ///  - matched but coming from known liquidity providers.
+    /// Should help us to identify how much we can save by parallelizing execution.
+    fn report_matched_orders(
         &self,
         submitted: &Settlement,
         all: impl Iterator<Item = RatedSettlement>,
+        orders: Vec<LimitOrder>,
     ) {
         let submitted: HashSet<_> = submitted
             .trades()
             .iter()
             .map(|trade| trade.order.order_meta_data.uid)
             .collect();
-        let all_matched: HashSet<_> = all
+        let all_matched_orders: HashSet<_> = all
             .flat_map(|solution| solution.settlement.trades().to_vec())
-            .map(|trade| trade.order.order_meta_data.uid)
+            .map(|trade| trade.order)
+            .collect();
+        let all_matched_ids: HashSet<_> = all_matched_orders
+            .iter()
+            .map(|order| order.order_meta_data.uid)
+            .collect();
+        let matched_but_not_settled = all_matched_ids.difference(&submitted).copied().collect();
+        let liquidity_order_ids: HashSet<_> = orders
+            .into_iter()
+            .filter_map(|order| match order.is_liquidity_order {
+                true => Some(
+                    OrderUid::from_str(&order.id).expect("LimitOrder was constructed from Order"),
+                ),
```

Comment:
This might cause unexpected panics if we add limit orders from other sources in the future (for example 0x). Might be worth to just filter out failures:
```suggestion
                true => OrderUid::from_str(&order.id).ok(),
```